draft-schedulers.mkd

```diff
@@ -197,25 +217,67 @@ Segment Size that is negotiated at connection establishment.
 
 [comment]: # (MA: Linking send and rcv window to a reliable protocol is misleading for MP-QUIC and Datagram or MP-DCCP. Also do not considering path level and linking to connection level only will prevent any path scheduling)
 
+
 It may also contain some information that is specific to the
 application (e.g. total amount of data sent or received) and information about
 non-active flows. At the path level, we expect that the multipath transport
 protocol will maintain a accurate estimation of the round-trip-time over that
 path, possibly a send/receive window, per path MTU information, the state of the
 congestion controller, and optionally information that is specific to the
 application or the packet scheduler (e.g. priority for one path over another
-one). 
+one).
 
 [comment]: # (To be continued and updated)
 
 # Conventions and Definitions
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
 "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
-"OPTIONAL" in this document are to be interpreted as described in BCP 14 
+"OPTIONAL" in this document are to be interpreted as described in BCP 14
 {{RFC2119}}{{RFC8174}} when, and only when, they appear in all capitals,
 as shown here.
 
+# Packet scheduling challenges {#challenges}
+
+Packet scheduling tries to balance different quality of service goals with
+different constraints of the paths. The balance depends on which of the goals
+or constraints is the primary factor for the experience the application is
+aiming for. In the following we list these goals and constraints and conclude
+by how they can influence each other.
+
+Each path can be subject to a different cost when transmitting data. For example,
+a path can introduce a per-byte monetary cost for the transmission (e.g., metered
+cellular link). Another cost can be the power consumption when transmitting or
+receiving data. These costs are imposing restrictions on when a path can be used
+compared to the lower-cost path.
+
+A goal for many applications is to reduce the latency of their transaction. With
+multiple paths, each path can have a significantly different latency compared to
+the other paths. It is thus crucial to schedule the traffic on a path such that
+the latency requirements of the application are satisfied.
+
+Achieving high throughput is another goal of many applications. Streaming applications
+often require a minimum bitrate to sustain playback. The scheduler should try to
+achieve this bitrate to allow for a flawless streaming experience. Beyond that,
+adaptive streaming requires also a more stable throughput experience to ensure
+that the bitrate of the video stream is consistent.
+
+Finally, transport protocols impose a receive-window that signals to the sender
+how much data the application is willing to receive. When the paths have a large
```

Comment:
Can we assume that any transport protocol maintains a rcv window?

Reply:
otherwise there is a notion of maximum rate but if we try to be too generic the document will become unreadable

Reply:
In principle I agree, however I would  rephrase to become a little independent of the pretty specific rcv wnd. I rather propose to focus in this document on transport protocols which acknowledge (ACK) received data and derive a snd window thereof (congestion control).